socrata/resource.py

```diff
@@ -167,16 +169,28 @@ def show(self, uri):
             auth = self.auth
         ))
 
-
     def _wait_for_finish(self, is_finished, is_failed, progress, timeout, sleeptime):
-        started = time.time()
-        while not is_finished(self):
-            current = time.time()
-            if timeout and (current - started > timeout):
-                raise TimeoutException("Timed out after %s seconds waiting for completion for %s" % (timeout, str(self)))
-            me = self.show()
-            progress(self)
-            if is_failed(self):
-                raise ResourceFailedException(me)
-            time.sleep(sleeptime)
-        return self
+        def retry(e, attempts):
+            if attempts < 5:
+                attempts = attempts + 1
+                sleep(attempts * attempts * 2)
+                return wff(time.time(), attempts)
+            else:
+                raise e
+
+        def wff(started, attempts):
+            while not is_finished(self):
+                current = time.time()
+                if timeout and (current - started > timeout):
+                    timeoutWarn = "Timed out after %s seconds waiting for completion for %s on attempts %i" % (timeout, str(self), attempts)
+                    log.warn(timeoutWarn)
+                    ex = TimeoutException(timeoutWarn)
+                    retry(e, attempts)
```

Comment:
I don't see `e` being bound here?  Also, I think this one should just be raised, because this timeout is saying "you asked to poll for only `timeout` seconds, and we've reached that point without success."